simln-lib/src/lib.rs

```diff
@@ -321,20 +321,42 @@ pub trait PaymentGenerator: Display + Send {
     ) -> Result<u64, PaymentGenerationError>;
 }
 
+/// Represents the result of a payment attempt.
 #[derive(Debug, Clone, Serialize, Deserialize)]
 pub struct PaymentResult {
+    /// The number of HTLCs (Hash Time Locked Contracts) used in the payment attempt.
+    /// Multiple HTLCs may be used for a single payment when using techniques like
+    /// multi-part payments or when retrying failed payment paths.
     pub htlc_count: usize,
+    /// The final outcome of the payment attempt, indicating whether it succeeded
+    /// or failed (and if failed, the reason for failure).
     pub payment_outcome: PaymentOutcome,
 }
 
 impl PaymentResult {
+    /// Creates a new PaymentResult indicating that the payment was never dispatched.
+    /// This is used when there was an error during the initial payment dispatch attempt.
+    ///(e.g., insufficient balance, invalid destination)
+    ///
+    /// # Returns
+    /// A PaymentResult with:
+    /// - htlc_count: 0 (no HTLCs used since payment wasn't dispatched)
+    /// - payment_outcome: NotDispatched
     pub fn not_dispatched() -> Self {
         PaymentResult {
             htlc_count: 0,
             payment_outcome: PaymentOutcome::NotDispatched,
         }
     }
 
+    /// Creates a new PaymentResult indicating that tracking the payment failed.
+    /// This is used when the payment was dispatched but the system was unable to
+    /// determine its final outcome (e.g., due to connection issues or timeouts).
+    ///
+    /// # Returns
+    /// A PaymentResult with:
+    /// - htlc_count: 0 (unknown since tracking failed)
+    /// - payment_outcome: TrackPaymentFailed
```

Comment:
I think we can shorten these returns sections to just be included in the top line, then remove the "returns" section:

`Creates a new PaymentResult indicating that tracking the payment failed using [PaymentOutcome::TrackPaymentFailed]`

We're probably going to want to turn on the following to make sure we keep the good habit of docs up to date + don't let code vars get out of date:
`#![deny(rustdoc::broken_intra_doc_links)]`
`#![deny(missing_docs)]`

So let's get the references to code vars right here, eg: [\`PaymentOutcome::TrackPaymentFailed\`]